lldb/source/API/SBThread.cpp

```diff
@@ -859,7 +859,9 @@ SBError SBThread::StepOverUntil(lldb::SBFrame &sb_frame,
       addr_t step_addr =
           sc.line_entry.range.GetBaseAddress().GetLoadAddress(target);
       if (step_addr != LLDB_INVALID_ADDRESS) {
-        if (fun_range.ContainsLoadAddress(step_addr, target))
+        if (llvm::any_of(fun_ranges, [&](const AddressRange &r) {
```

Comment:
Why isn't there an AddressRanges::ContainsLoadAddress?  Seems silly to have to write this loop by hand.

Reply:
AddressRanges is a typedef for vector<AddressRange>.

How about using `Block::GetRangeContainingLoadAddress`  (Function::GetAddressRanges just returns the data from the function Block) for this instead (see new version of the patch)? Other possibilities might be:
- add Function::GetRangeContainingLoadAddress or Function::ContainsLoadAddress
- add a free function (`ContainsLoadAddress(ArrayRef<AddressRange>, addr_t, Target)`)

Reply:
It feels awkward to have to manually dial up the function block to see if an address exists in that function.  I'd vote for `Function::GetRangeContainingLoadAddress` - that just seems like something a Function should be able to tell you.

Reply:
SG